### PR TITLE
Don't mount /dev/shm if the user told you --ipc=none

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: libpod/container_internal.go [hunk 1]
```diff
@@ -914,8 +914,10 @@ func (c *Container) makeBindMounts() error {
 		c.state.BindMounts = make(map[string]string)
 	}
 
-	// SHM is always added when we mount the container
-	c.state.BindMounts["/dev/shm"] = c.config.ShmDir
+	if c.config.ShmDir != "none" {
```
Comment (mheon) [inline libpod/container_internal.go]: Instead of this - we should probably check the spec for an IPC namespace, I think

Comment (rhatdan) [reply libpod/container_internal.go]: I am not sure what you mean?

Comment (mheon) [reply libpod/container_internal.go]: The namespaces the container is using are listed in the OCI spec. If there is no IPC namespace listed, that means that we have an IPC=host container, so we don't need to pass in a special keyword to tell libpod not to make an SHM mount

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**